### DISCUSSION TITLE
specify region in route53

### DISCUSSION THREAD

**Discussion by demimike** [gruntwork-io/knowledge-base#606] [How do I] [answered]
I would like to use route 53 to create a cert in us-east-1 region since we will be using it with cloudfront to private s3. It seems to use our default region us-west-2. Is there a way to override the region in this module? 

---

<ins datetime="2022-12-02T21:20:53Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109687">Tracked in ticket #109687</a></p>
</ins>

**Comment by Etiene** [comment, marked as answer]
Hi @demimike, in our `load-balancer` repo we have an [`acm-tls-certificate` module](https://github.com/gruntwork-io/terraform-aws-load-balancer/tree/main/modules/acm-tls-certificate), here's a usage [example](https://github.com/gruntwork-io/terraform-aws-load-balancer/blob/e7bd8207733c3bd71c86d7aed05d79f542c4e8ad/examples/acm-tls-certificate/main.tf). You can set the region in the provider block. Is this what you mean?

**Reply by demimike** [reply]
We are using the reference architecture. In that structure, the route53-public.hcl is in the /dev/_global folder. If I need to have route53-public run in region us-east-1 but our default region is us-west2, how to make this happen?

**Reply by Etiene** [reply]
Oh I see! It seems that, according [this issue](https://github.com/gruntwork-io/terraform-aws-service-catalog/issues/328), configuring the certificate in other regions is not supported by the route53 networking module yet. I can see two possible ways of going around it:

1. Creating a /dev/us-east-1 folder, with its own region.hcl configured for us-east-1, create a new [child configuration](https://github.com/gruntwork-io/terraform-aws-service-catalog/blob/master/core-concepts.md#deploy-new-infrastructure) in it, and call the acm-tls-certificate module directly just for the creation of the extra certificate.
```
include {
  path = find_in_parent_folders()
}
terraform {
  source = "git::git@github.com:gruntwork-io/terraform-aws-load-balancer.git//modules/acm-tls-certificate?ref=<VERSION>"
}

inputs = {
  <INPUTS>
}
```

or 

2. Creating a /dev/us-east-1 folder, with its own region.hcl configured for us-east-1, and moving /dev/_global/route53-public to /dev/us-east-1/us-east-1
